packages/asset-uploader/src/schema.ts

```diff
@@ -43,7 +43,7 @@ const BaseAsset = z.object({
   location: Location,
   createdAt: z.string(),
   path: z.string(),
-  status: z.literal("uploaded").optional(),
```

Comment:
This is a transform right? In other words this is possible now `equal(x, BaseAsset.parse(x)) === false`?

I think we should not use transforms in Zod types that are used to define TypeScript types. This is very confusing to me, because now we have two types to think about: the input type and the output type. Just one of the many questions:

Which of the two we store data in DB in?

If in the output type then we can't parse when reading from DB: `BaseAsset.parse(dataFromDatabaseInOutputFormat)` — wrong.

If in the input format, writing to DB would have to look something like this:
`const dontUsetThis = BaseAsset.parse(dataFromClientInInputFormat); db.write(dataFromClientInInputFormat)`

Reply:
`equal(x, BaseAsset.parse(x)) === false` not possible as `status` now not optional type

Reply:
The issue is that optional `updated` makes code much harder. Making it default will not break anything but make types much more easier to use.

Reply:
Talked in dm. Takeaway:

1. Transforms in shared reusable Zod types are bad because you have to think about `In` and `Out` types.
2. But if `Out extends In` it's not as bad, because at least you can still validate data in both formants using the `Zod` schema. This is the case here, so tolerable.

Reply:
Here is the other issue, that field `status:"uploaded"` has no sense at server data at all. And seems like occurred there by mistake. Because types are reused, it was added in typescript type, then ts type was transformed into Zod. 
I narrowed it down from optional to required. Making the issue even worse.

Reply:
agree, yes, it was supposed to be only for the client

Reply:
Removed